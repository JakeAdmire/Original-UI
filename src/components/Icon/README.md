# Icon

Need a quick and painless icon? Easy! Need the icon to function as a pressable button? Just as easy! No more complex SVG editors or giant blocks of code for you, just import Icon!

**Original-UI** utilizes [Font Awesome](fontawesome.com) to build it's icons. To take advantage of this component you must first follow the instructions at [fontawesome.com](fontawesome.com) and implement the `<script>` tag within the enclosing `<head>` tags of your `index.html` file.


## Example Usage:

  
### Simple Icons with defined colors:
```javascript
import { Icon } from  'original-ui'

<Icon
	iconName='skull'
	iconColor='white'
	backgroundColor='red'/>
<Icon
	iconName='bath'
	iconColor='white'
	backgroundColor='cornflowerblue'  />
<Icon
	iconName='bomb'
	iconColor='white'
	backgroundColor='black'  />
```


**Result:**

<img  width="205"  alt="Screen Shot 2019-10-20 at 5 47 46 PM"  src="https://user-images.githubusercontent.com/44077214/67168473-22574500-f362-11e9-8c02-961901f03072.png">

## Parameters:

|Parameter| Data Type | Default Value | Description |
|--|--|--|--|
| `iconName` | `String` | `null` | Pass a string with the name of a Font Awesome icon to build the icon |
| `iconType` | `String` | `'Solid'` | _Pass a string of 'Solid', 'Regular', 'Light', 'Duotone', or 'Brands' to determine the icon style. Only 'Solid' & 'Brands' are included by Font Awesome for free_ |
| `iconSize` | `String` | `'24px'` | _Pass a string to constrict the icon's size_ |
| `iconColor` | `String` | `'#00000033'` | _Pass a string to set the icon's color_ |
| `rotation` | `String` | `null` | _Pass a string of '90', '180', or '270' to rotate the icon to that degree_ |
| `isFlippedVertically` | `Boolean` | `false` | _Pass a boolean to determine whether the icon is flipped vertically_ |
| `isFlippedHorizontally` | `Boolean` | `false` | _Pass a boolean to determine whether the icon is flipped horizontally_ |
| `isSpinning` | `Boolean` | `false` | _Pass a boolean to determine whether the icon spins continuously_ |
| `isPulsing` | `Boolean` | `false` | _Pass a boolean to determine whether the icon pulses continuously_ |
| `onPress` | `Function` | `null` | _Pass a function to be called on icon press. Icon will not behave as a button if `onPress` is null_ |
| backgroundColor | `String` | `null` | _Pass a string to set the icon's background color_ |