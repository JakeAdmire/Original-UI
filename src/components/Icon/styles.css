.iconwrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  /**/
  border-radius: 50px;
  position: relative;
}

.iconwrapper:hover::after {
  background: #80808040;
  cursor: pointer;
}

.iconbutton:active {
  transition: opacity .3s ease;
  opacity: 0.5;
}

.iconwrapper::after {
  content: '';
  transition: all .3s ease;
  border-radius: 50px;
  /**/
  position: absolute;
  top: 0; right: 0; left: 0; bottom: 0;
}

.icon {
  box-sizing: border-box;
}