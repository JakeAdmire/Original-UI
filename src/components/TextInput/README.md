# TextInput

Need to make a text input of some kind? An easily style-able search bar, or some login fields? **TextInput** is your component! Simply import `TextInput` and use the [parameters](#parameters) to start customizing, or you can copy and paste directly from one of my examples below!

## Example Usage:

### Simple Search Bar:
```javascript
<TextInput
	placeholderText='Start typing to search...'
	onChange={this.handleSearchChange}
	leftIcon={<Icon iconName={'search'}/>}/>
```
**Result:**

<img width="838" alt="Screen Shot 2019-10-21 at 2 49 58 PM" src="https://user-images.githubusercontent.com/44077214/67242483-dc0dee80-f412-11e9-9d99-982c055e2a28.png">

---


### Password input with hidden value toggle:
```javascript
import { TextInput } from  'original-ui'

<TextInput 
  placeholderText='ex. "thoughtfulpassword123..."'
  labelText='Enter your password:' 
  isPassInput={this.state.isTextHidden}
  rightIcon={
    <Icon 
      iconName={this.state.isTextHidden ? 'eye' : 'eye-slash'}
      iconSize='24px'
      onPress={this.toggleHiddenText}/>
  }/>
```

**Result:**

<img width="838" alt="Screen Shot 2019-10-20 at 9 14 45 PM" src="https://user-images.githubusercontent.com/44077214/67174278-b931fa80-f37e-11e9-8e9c-fe374a3642cb.png">


## Parameters:

|Parameter| Data Type | Default Value | Description |
|--|--|--|--|
| `placeholderText` | `String` | `null` | _Pass a string to display as a placeholder_ |
| labelText | `String` | `null` | _Pass a string to display as a label_ |
| labelStyle | `Object` | `{ color: 'orange' }` | _Pass a style object to set the label's style rules_ |
| isPassInput | `Boolean` | `false` | _Pass a boolean to determine whether you want the input's text obscured_ |
| leftIcon | `Component` | `null` | _Pass an [Icon](#icon) component to display on the left side of the input_ |
| rightIcon | `Component` | `null` | _Pass an [Icon](#icon) component to display on the right side of the input_ |
| `width` | `String` | `'100%'` | _Pass a string to constrict the component's width_ |
| `height` | `String` | `'50px'` | _Pass a string to constrict the component's height_ |
| `textStyle` | `Object` | `{ fontSize: '16px', color: 'grey' }` | _Pass a style object to set the input text's style rules_ |
| `onChange` | `Function` | `null` | _Pass a method that will receive the `event` object as a parameter_ |
| `value` | 'String' | `null` | _Pass a string to set the input's value_ |
---