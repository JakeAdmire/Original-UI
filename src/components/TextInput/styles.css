.inputlabel {
  font-weight: 300;
}

.inputwrapper {
  box-sizing: border-box;
  padding: 6px;
  display: flex;
  align-items: center;
  transition: all .3s ease;
  background: #e7e7e7;
  border-radius: 50px;
}

.textinput {
  box-sizing: border-box;
  padding: 0 10px;
  outline: none;
  width: 100%;
  height: 100%;
  border: none;
  background: none;
  transition: all .3s ease;
}

.textinput::placeholder {
  color: #00000033;
}

.inputshadow {
  box-shadow: inset 0 0 12px 0px rgba(0, 0, 0, 0.15);
}