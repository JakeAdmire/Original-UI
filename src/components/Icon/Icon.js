import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

import { isRequired, tooManyParameters } from '../../helpers/helpers'

export class Icon extends Component {
  static propTypes = {
    iconName: PropTypes.string,
    iconType: PropTypes.string,
    iconSize: PropTypes.string,
    iconColor: PropTypes.string,
    rotation: PropTypes.string,
    isFlippedVertically: PropTypes.bool,
    isFlippedHorizontally: PropTypes.bool,
    isSpinning: PropTypes.bool,
    isPulsing: PropTypes.bool,
    onPress: PropTypes.func,
    backgroundColor: PropTypes.string
  };

  render() {
    const {
      iconName = null,
      iconType = 'Solid',
      iconSize = '24px',
      iconColor = '#00000033',
      rotation = null,
      isFlippedVertically = false,
      isFlippedHorizontally = false,
      isSpinning = false,
      isPulsing = false,
      onPress,
      backgroundColor
    } = this.props

    iconName === null && isRequired('Name', 'Icon')
    if (isFlippedVertically && isFlippedHorizontally) tooManyParameters('Icon', ['isFlippedVertically', 'isFlippedHorizontally'])
    if (isSpinning && isPulsing) tooManyParameters('Icon', ['isSpinning', 'isPulsing'])

    let iconRotate = rotation !== null && (rotation === '90' || '180' || '270')
      ? `fa-rotate-${rotation}`
      : null

    let iconFlip
    if (isFlippedVertically) iconFlip = 'fa-flip-vertical'
    if (isFlippedHorizontally) iconFlip = 'fa-flip-horizontal'

    let iconSpin
    if (isSpinning) iconSpin = 'fa-spin'
    if (isPulsing) iconSpin = 'fa-pulse'

    let iconClass = `
      ${styles.icon} 
      fa${iconType.toLowerCase().split('')[0]} 
      fa-${iconName} 
      ${iconRotate} 
      ${iconFlip} 
      ${iconSpin} 
      ${onPress && styles.iconbutton}
    `

    let iconInlineStyles = {
      fontSize: iconSize,
      color: iconColor
    }

    let wrapperClass = `
      ${styles.iconwrapper} 
      ${onPress && styles.iconbutton}
    `

    let wrapperInlineStyles = {
      backgroundColor,
      height: Math.floor(Number(iconSize.match(/[0-9]+/g)[0]) * 1.7),
      width: Math.floor(Number(iconSize.match(/[0-9]+/g)[0]) * 1.7)
    }

    return (
      <div
        className={wrapperClass}
        style={wrapperInlineStyles}
        onMouseUp={onPress}>
        <i
          className={iconClass}
          style={iconInlineStyles} />
      </div>
    )
  }
}
