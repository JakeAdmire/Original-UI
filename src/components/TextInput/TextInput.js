import React, { Component } from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

export class TextInput extends Component {
  static propTypes = {
    placeholder: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
    fontsize: PropTypes.string,
    showShadow: PropTypes.bool,
    border: PropTypes.string,
    radius: PropTypes.string,
    vanishingPlaceholder: PropTypes.bool,
    changeMethod: PropTypes.func
  };

  render() {
    const {
      placeholder,
      width = '100%',
      height = '50px',
      fontsize = '16px',
      showShadow = true,
      border = 'orangered',
      radius = '10px',
      vanishingPlaceholder = true,
      changeMethod
    } = this.props

    let inlineStyles = {
      width: width,
      height: height,
      fontSize: fontsize,
      borderColor: border,
      borderRadius: radius
    }

    let inputClass = `
      ${styles.textinput} 
      ${showShadow && styles.inputshadow}
      ${vanishingPlaceholder && styles.vanishingPlaceholder}
    `

    return (
      <input
        className={inputClass}
        placeholder={placeholder || null}
        style={inlineStyles}
        onChange={changeMethod !== null ? (e) => changeMethod(e) : null}
      />
    )
  }
}
