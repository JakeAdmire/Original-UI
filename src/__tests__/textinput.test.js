/* eslint-disable padded-blocks */
import React from 'react'
import { shallow } from 'enzyme'

import { TextInput } from '../components/TextInput/TextInput'
import { Icon } from '../components/Icon/Icon'

describe('TextInput', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <TextInput />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('[Parameter] placeholderText', () => {

    afterEach(() => {
      wrapper.setProps({ plaeholderText: '' })
    })

    it('should have a default value', () => {
      let input = wrapper.find('input')
      expect(input.props().placeholder).toEqual('')
    })

    it('should give TextInput placeholder reflecting parameter', () => {
      expect(wrapper.find('input').props().placeholder).toEqual('')

      let placeholderText = 'Start typing to search...'
      wrapper.setProps({ placeholderText: placeholderText })

      expect(wrapper.find('input').props().placeholder).toEqual(placeholderText)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

  describe('[Parameter] labelText', () => {

    afterEach(() => {
      wrapper.setProps({ labelText: undefined })
    })

    it('should not have a default value', () => {
      let label = wrapper.props().children[0]
      expect(label).toEqual(undefined)
    })

    it('should give TextInput a label reflecting parameter', () => {
      let labelText = 'Enter your email:'

      wrapper.setProps({ labelText })

      let labelInnerText = wrapper.find('label').props().children
      expect(labelInnerText).toEqual(labelText)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

  describe('[Parameter] labelStyle', () => {

    afterEach(() => {
      wrapper.setProps({ labelStyle: `{ color: 'orange' }` })
    })

    it('should have a default value', () => {
      let labelStyle = { color: 'orange' }

      wrapper.setProps({ labelText: 'Enter your email' })

      let label = wrapper.find('label')
      expect(label.props().style).toEqual(labelStyle)

    })

    it('should give TextInput label style reflecting parameter', () => {
      let labelStyle = `{ fontSize: '34px', color: 'blue' }`

      wrapper.setProps({ 
        labelText: 'Email:',
        labelStyle
      })

      let label = wrapper.find('label')
      expect(label.props().style).toEqual(labelStyle)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

  describe('[Parameter] width', () => {

    afterEach(() => {
      wrapper.setProps({ width: '100%' })
    })

    it('should have a default value', () => {
      let width = '100%'

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().style.width).toEqual(width)
    })

    it('should give TextInput style reflecting parameter', () => {
      let width = '50px'

      wrapper.setProps({ width })

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().style.width).toEqual(width)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

  describe.skip('[Parameter] height', () => {

    afterEach(() => {
      wrapper.setProps({ height: '50px' })
    })

    it('should have a default value', () => {
      let height = '50px'

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().style.height).toEqual(height)
    })

    it('should give TextInput style reflecting parameter', () => {
      let height = '100%'

      wrapper.setProps({ height })

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().style.height).toEqual(height)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

  describe('[Parameter] textStyle', () => {

    afterEach(() => {
      wrapper.setProps({ textStyle: `{
        fontSize: '16px',
        color: 'grey'
      }` })
    })

    it('should have a default value', () => {
      let textStyle = {
        fontSize: '16px',
        color: 'grey'
      }
      let input = wrapper.find('input')
      expect(input.props().style).toEqual(textStyle)
    })

    it('should give TextInput style reflecting parameter', () => {
      let textStyle = `{
        fontWeight: 800
      }`

      wrapper.setProps({ textStyle })

      let input = wrapper.find('input')
      expect(input.props().style).toEqual(textStyle)
    })

    it.skip('should throw an error if parameter is not type:Object')

  })

  describe('[Parameter] onChange', () => {

    afterEach(() => {
      wrapper.setProps({ onChange: null })
    })

    it('should not have a default value', () => {
      let input = wrapper.find('input')

      expect(input.props().onChange).toEqual(null)
    })

    it('should be called on TextInput change', () => {
      let onChange = jest.fn()

      wrapper.setProps({ onChange })

      let input = wrapper.find('input')
      input.simulate('change')

      expect(onChange).toBeCalled()
    })

    it.skip('should throw an error if parameter is not type:Function', () => {

    })

  })

  describe('[Parameter] isPassInput', () => {

    afterEach(() => {
      wrapper.setProps({ isPassInput: false })
    })

    it('should have a default value', () => {
      let input = wrapper.find('input')

      expect(input.props().type).toEqual('text')
    })

    it('should give TextInput input type reflecting parameter', () => {
      let isPassInput = true

      wrapper.setProps({ isPassInput })

      let input = wrapper.find('input')
      expect(input.props().type).toEqual('password')
    })

    it.skip('should throw an error if parameter is not type:Boolean', () => {

    })

  })

  describe('[Parameter] leftIcon', () => {

    afterEach(() => {
      wrapper.setProps({ leftIcon: undefined })
    })

    it('should not have a default value', () => {
      let inputWrapper = wrapper.find('.inputWrapper')

      expect(inputWrapper.props().children[0]).toEqual(undefined)
    })

    it('should give TextInput element reflecting parameter', () => {
      let leftIcon = <Icon iconName='search' />

      wrapper.setProps({ leftIcon })

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().children[0]).toEqual(leftIcon)
    })

    it.skip('should throw an erorr if parameter is not type:??Element??', () => {

    })

  })

  describe('[Parameter] rightIcon', () => {

    afterEach(() => {
      wrapper.setProps({ rightIcon: undefined })
    })

    it('should not have a default value', () => {
      let inputWrapper = wrapper.find('.inputWrapper')

      expect(inputWrapper.props().children[2]).toEqual(undefined)
    })

    it('should give TextInput element reflecting parameter', () => {
      let rightIcon = <Icon iconName='search' />

      wrapper.setProps({ rightIcon })

      let inputWrapper = wrapper.find('.inputWrapper')
      expect(inputWrapper.props().children[2]).toEqual(rightIcon)
    })

    it('should throw an error if parameter is not type:??Element??', () => {

    })

  })

  describe('[Parameter] value', () => {

    afterEach(() => {
      wrapper.setProps({ value: undefined })
    })

    it('should not have a default value', () => {
      let input = wrapper.find('input')

      expect(input.props().value).toEqual(undefined)
    })

    it('should give TextInput input value reflecting parameter', () => {
      let value = 'applesauce'

      wrapper.setProps({ value })

      let input = wrapper.find('input')
      expect(input.props().value).toEqual(value)
    })

    it.skip('should throw an error if parameter is not type:String', () => {

    })

  })

})
