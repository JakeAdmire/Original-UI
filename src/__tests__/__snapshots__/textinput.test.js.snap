// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="
      undefined
      false
    "
          style={
                    Object {
                              "height": "50px",
                              "width": "100%",
                            }
          }
>
          <input
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    style={
                              Object {
                                        "color": "grey",
                                        "fontSize": "16px",
                                      }
                    }
                    type="text"
                    value={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <input
              className={undefined}
              id={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              style={
                            Object {
                                          "color": "grey",
                                          "fontSize": "16px",
                                        }
              }
              type="text"
              value={undefined}
/>,
            undefined,
          ],
          "className": "
      undefined
      false
    ",
          "style": Object {
            "height": "50px",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "id": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "",
              "style": Object {
                "color": "grey",
                "fontSize": "16px",
              },
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="
      undefined
      false
    "
            style={
                        Object {
                                    "height": "50px",
                                    "width": "100%",
                                  }
            }
>
            <input
                        className={undefined}
                        id={undefined}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        style={
                                    Object {
                                                "color": "grey",
                                                "fontSize": "16px",
                                              }
                        }
                        type="text"
                        value={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <input
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder=""
                style={
                                Object {
                                                "color": "grey",
                                                "fontSize": "16px",
                                              }
                }
                type="text"
                value={undefined}
/>,
              undefined,
            ],
            "className": "
      undefined
      false
    ",
            "style": Object {
              "height": "50px",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "id": undefined,
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "placeholder": "",
                "style": Object {
                  "color": "grey",
                  "fontSize": "16px",
                },
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
